lib/file-view.coffee

```diff
@@ -19,7 +19,11 @@ class FileView extends HTMLElement
     @fileName.dataset.name = @file.name
     @fileName.dataset.path = @file.path
 
-    @fileName.classList.add(FileIcons.getService().iconClassForPath(@file.path))
+    iconClass = FileIcons.getService().iconClassForPath(@file.path, this)
```

Comment:
> After a quick glance, it doesn't seem to have any reference to a File object,

Here:
```coffee
TabView.item?.buffer.file
```

> What I am worried about is inconsistency, though

To be honest, inconsistency is a trivial sacrifice to make until Atom has a formal API for interacting with the filesystem. And yes, you're 100% right. Furthermore, I believe *that* API should be the one consuming the icon-class service... that way, there's only one point of info, and services can deal with that without having to care about where the class is being used.